src/main/webapp/WEB-INF/web.xml

```diff
@@ -32,9 +32,4 @@
     <url-pattern>/products/*</url-pattern>
   </servlet-mapping>
 
```

Comment:
The approach with using <error-page> is the correct way.
Please remove try-catch(ProductNotFoundException e) in servlet and use error-page, just please make sure error page shows friendly user message, not stack trace.